code/modules/research/nanites/nanite_programs/utility.dm

```diff
@@ -356,3 +356,19 @@
 
 /datum/action/innate/nanite_button/Activate()
 	program.press()
+
+/datum/nanite_program/change_cloud
+	name = "Change Cloud"
+	desc = "When triggered, changes the nanites' cloud of reference."
+	unique = FALSE
+	can_trigger = TRUE
+	trigger_cost = 0
+	trigger_cooldown = 10
+	rogue_types = list(/datum/nanite_program/glitch)
+	
+/datum/nanite_program/change_cloud/register_extra_settings()
+	extra_settings[NES_CLOUD_OVERWRITE] = new /datum/nanite_extra_setting/number(0, 0, 100)
+
+/datum/nanite_program/change_cloud/on_trigger(comm_message)
+	var/datum/nanite_extra_setting/cloud = extra_settings[NES_CLOUD_OVERWRITE]
+	SEND_SIGNAL(host_mob, COMSIG_NANITE_SET_CLOUD, cloud.get_value() )
```

Comment:
Signals need to happen at the base of the proc.

Reply:
I see, replacing SEND_SIGNAL it with the set_cloud proc.

Reply:
This in still unaddressed.

Reply:
This should be addressed now.